### ISSUE TITLE
empty files in run_exp/results/log_RL...

### ISSUE THREAD

**Issue by Ashling123** [hongzimao/pensieve#98]
While I conduct the experiment on mahimahi, I try the command “ python run_traces.py  ../cooked_traces RL 7 ip” . It  doesn’t show any errors in chrome_retry_log, however, the log files generated in run_exp/results are empty. Can you give me some advices about this ? By the way, I conduct the experiment in the Virtual machine with Selinium 2.39.0 , chrome 78.0, and the right chromedriver. And I use py3.5. I‘ll be really appreciate it if you can give me some help : ) Thanks!!!

**Comment by Ashling123** [comment]
And I have added the code https://github.com/park-project/park/blob/master/park/envs/abr/run_video.py#L63 , And it still got empty content.

**Comment by hongzimao** [comment]
Wow really nice that you scouted our other repo to find that fix - that's what I would have suggested :)

With the setup, you should be able to 1. run rl_server_no_training (or other server in https://github.com/hongzimao/pensieve/tree/master/rl_server), 2. start web server (that should be by default when you installed apache2, 3. start a chrome browser and go to localhost/myindex_RL.html (or other html in https://github.com/hongzimao/pensieve/tree/master/video_server). This bypasses the mahimahi and you can see if the video loads properly (apologize if the video content is not the most pleasant one!).

The idea is basically runs the command in the automated script and see where things get stuck. Hope this helps!

**Comment by Ashling123** [comment]
I’m really grateful for your emails. And with the advices you gives to me, I fixed some problems caused by the type error so that I can load the videos properly when bypassing the mahimahi. And the videos look artistic. However, the problem of empty files in results still exists, and it seems that when using command to run ‘run_traces.py’, ‘rl_server_no_training.py’ doesn’t work.  Are there something wrong with the command of ‘proc’ in ‘run_video.py’? I’m still confused with it. Thanks for your reply again. : )------------------ 原始邮件 ------------------
发件人: "Hongzi Mao"<notifications@github.com>
发送时间: 2020年4月30日(星期四) 凌晨4:00
收件人: "hongzimao/pensieve"<pensieve@noreply.github.com>;
抄送: "Ashling123"<154117377@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [hongzimao/pensieve] empty files in run_exp/results/log_RL...(#98)




 
Wow really nice that you scouted our other repo to find that fix - that's what I would have suggested :)
 
With the setup, you should be able to 1. run rl_server_no_training (or other server in https://github.com/hongzimao/pensieve/tree/master/rl_server), 2. start web server (that should be by default when you installed apache2, 3. start a chrome browser and go to localhost/myindex_RL.html (or other html in https://github.com/hongzimao/pensieve/tree/master/video_server). This bypasses the mahimahi and you can see if the video loads properly (apologize if the video content is not the most pleasant one!).
 
The idea is basically runs the command in the automated script and see where things get stuck. Hope this helps!
 
&mdash;

**Comment by hongzimao** [comment]
You can try executing the command in https://github.com/hongzimao/pensieve/blob/1120bb173958dc9bc9f2ebff1a8fe688b6f4e93c/run_exp/run_video.py#L62 directly in your command line and see if it works.

(Just saw #100 successfully runs the experiment very recently. Might be worthwhile checking there to see the setup as well.)

**Comment by Ashling123** [comment]
I’m sorry to respond your email late, because I have some problems in my virtual machines and spend some time checking over the software configurations. I have solve the problem with your suggestion, and I’m really happy and grateful ! ^ ^ Now I want to get more figures that show pensieve’s advantages. Could you tell me some details about getting the figure 8 and 9 in your paper ? Besides changing HD_reward in multi_agent.py and retrain a new model, what else do I need to modify? Thanks for your replying: ) ------------------ 原始邮件 ------------------
发件人: "Hongzi Mao"<notifications@github.com>
发送时间: 2020年4月30日(星期四) 晚上8:08
收件人: "hongzimao/pensieve"<pensieve@noreply.github.com>;
抄送: "Ashling123"<154117377@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [hongzimao/pensieve] empty files in run_exp/results/log_RL...(#98)




 
You can try executing the command in https://github.com/hongzimao/pensieve/blob/1120bb173958dc9bc9f2ebff1a8fe688b6f4e93c/run_exp/run_video.py#L62 directly in your command line and see if it works.
 
(Just saw #100 successfully runs the experiment very recently. Might be worthwhile checking there to see the setup as well.)
 
&mdash;

**Comment by hongzimao** [comment]
Glad to hear you made it work! I remember the reward signal was pretty much the only thing we changed in that evaluation. You might also want to consider using larger dataset for evaluation (see if you can download more data from the source (e.g., http://home.ifi.uio.no/paalh/dataset/hsdpa-tcp-logs/). We experienced better model performance with more data.

**Comment by Ashling123** [comment]
^ ^ ok, I've got what your said ~ And do I need to change the reward expression in MPC or other algorithms? I’ve thought this is a kind of QoE goal so other algorithms also need to achieve. Am I thought right? thanks for reply again. ------------------ 原始邮件 ------------------
发件人: "Hongzi Mao"<notifications@github.com>
发送时间: 2020年5月4日(星期一) 晚上9:24
收件人: "hongzimao/pensieve"<pensieve@noreply.github.com>;
抄送: "Ashling123"<154117377@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [hongzimao/pensieve] empty files in run_exp/results/log_RL...(#98)




 
Glad to hear you made it work! I remember the reward signal was pretty much the only thing we changed in that evaluation. You might also want to consider using larger dataset for evaluation (see if you can download more data from the source (e.g., http://home.ifi.uio.no/paalh/dataset/hsdpa-tcp-logs/). We experienced better model performance with more data.
 
&mdash;

**Comment by hongzimao** [comment]
Yes, you change all algorithms to the same reward if you need to compare them on the same ground.

**Comment by Ashling123** [comment]
Alright, Thank you 😊 ------------------ 原始邮件 ------------------
发件人: "Hongzi Mao"<notifications@github.com>
发送时间: 2020年5月5日(星期二) 上午9:57
收件人: "hongzimao/pensieve"<pensieve@noreply.github.com>;
抄送: "Ashling123"<154117377@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [hongzimao/pensieve] empty files in run_exp/results/log_RL...(#98)




 
Yes, you change all algorithms to the same reward if you need to compare them on the same ground.
 
&mdash;
